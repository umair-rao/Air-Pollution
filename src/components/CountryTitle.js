import React from 'react';
import PropTypes from 'prop-types';
import { FaArrowRight } from 'react-icons/fa';
import '../styles/CountryTitle.css';

const CountryTitle = ({ data }) => (
  <div className="airData">
    <div className="container">
      <a href={`/details/${data?.country}`} className="next-icon">
        <FaArrowRight />
      </a>
      <div className="details">
        <div className="card-title">
          <h2>{data?.country}</h2>
        </div>
        {
          data?.timeline && (
          <div className="vaccine">
            <p>{data.timeline[Object.keys(data.timeline)[0]]}</p>
          </div>
          )
        }
      </div>
    </div>
  </div>
);

CountryTitle.propTypes = {
  data: PropTypes.oneOfType([PropTypes.object]).isRequired,
  // timeline: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default CountryTitle;
