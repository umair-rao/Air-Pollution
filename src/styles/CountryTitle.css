* {
  margin: 0;
  padding: 0;
}

.airData {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 20px;
}

.container {
  width: 15rem;
  height: 9rem;
  background-color: #7e685a;
  color: #fff;
}

.next-icon {
  margin-left: 85%;
  padding-top: 3%;
  font-size: 1.5rem;
}

.details {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin-right: 5%;
  font-size: 1.5rem;
}
