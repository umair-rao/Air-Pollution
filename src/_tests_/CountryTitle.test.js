import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import store from '../Redux/configureStore';
import CountryTitle from '../components/CountryTitle';

describe('CountryTitle', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <CountryTitle />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
