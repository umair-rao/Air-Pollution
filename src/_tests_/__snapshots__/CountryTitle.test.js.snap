// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryTitle should render correctly 1`] = `
<div
  className="countryData"
>
  <div
    className="container"
  >
    <a
      className="next-icon"
      href="/details/undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        />
      </svg>
    </a>
    <div
      className="details"
    >
      <div
        className="card-title"
      >
        <h3 />
      </div>
    </div>
  </div>
</div>
`;
